### ISSUE TITLE
Cannot flash my Xperia Z2 Tablet (SGP512) under Linux

### ISSUE THREAD

**Issue by murdegern** [Androxyde/Flashtool#189]
Hi, I tried to flash my old Xperia Z2 Tablet back to 23.5.A.1.291 under Linux (had CarbonOS before, tried to go to Lineage 18), but only get the message

ERROR - The bundle does not match the connected device

The complete log is attached. The bootloader is unlocked (obviously).
Since I am really, really sure this is an SGP512 (it even says so on th
[log.txt](https://github.com/Androxyde/Flashtool/files/9844979/log.txt)
e side), can I somehow skip this safety check?

Thanks for any help!
- Torsten

**Comment by murkl** [comment]
Hi, same here with SGP521. You can try to use this tool: https://github.com/munjeni/xflasher

**Comment by murdegern** [comment]
Thanks @murkl, I tried the xflasher, but it didn't change anything for me - first, it rejected the loader.sin because my device apparently found it invalid, but succeeded in flashing the loader from the SGP512 firmware I downloaded. Didn't seem to change much, though. BUT - you made me try harder to get it done with Flashtool!

The breakthrough was when I cloned flashtool, changed getDevInfo() in Flashtool/src/main/java/org/flashtool/flashsystem/S1Flasher.java to always return "SGP512" and got it compiled (this was the most complicated step, I'm really strange to the Java ecosystem). Then, I could just flash it with flashtool. Lineage 18 is now running, yay :-)
